lib/kitchen/lifecycle_hook/base.rb

```diff
@@ -0,0 +1,78 @@
+module Kitchen
+  class LifecycleHook
+    class Base
+      # @return [Kitchen::LifecycleHooks]
+      attr_reader :lifecycle_hooks
+
+      # return [String]
+      attr_reader :phase
+
+      # return [Hash]
+      attr_reader :hook
+
+      # @param lifecycle_hooks [Kitchen::LifecycleHooks]
+      # @param phase [String]
+      # @param hook [Hash]
+      def initialize(lifecycle_hooks, phase, hook)
+        @lifecycle_hooks = lifecycle_hooks
+        @phase = phase
+        @hook = hook
+      end
+
+      # return [void]
+      def run
+        raise NotImplementedError
+      end
+
+      # @return [TrueClass, FalseClass]
+      def should_run?
+        if !includes.empty?
+          includes.include?(platform_name)
+        elsif !excludes.empty?
+          !excludes.include?(platform_name)
+        else
+          true
+        end
+      end
+
+      # @return [Logger] the lifecycle hooks's logger
+      #   otherwise
+      # @api private
+      def logger
+        lifecycle_hooks.send(:logger)
+      end
+
+      private
+
+      # @return [Kitchen::Instance]
+      def instance
+        lifecycle_hooks.instance
+      end
+
+      # @return [Hash]
+      def config
+        lifecycle_hooks.send(:config)
+      end
+
+      # @return [Kitchen::StateFile]
+      def state_file
+        instance.state_file
```

Comment:
When I tested this code, I get an exception thrown on this line because `state_file` is a private method of Kitchen::Instance. I reached here by having a Kitchenfile with a basic top-level lifecycle hook:
```
lifecycle:
  pre_create:
  - local: echo ....... in pre create .........
```

Reply:
ahh... I haven't tested this at all at this point.

Well... using `#send` seems like a hack and probably not recommended, but what do you think could be done here? I'm not really sure given that I don't know test kitchen that well. In other words, I'm looking for some ideas here of the best way to resolve this issue.

Reply:
Well, I see three options, in order of my preference:
1. The `state_file` object is actually passed in to the initialize method of LifecyleHooks. Previously, it was repeatedly passed down until it reached where it was needed. It's not graceful, but that is the least disruptive approach.
2. Use `#send` to call `state_file`
3. Alter Instance to make state_file public. I'm most reluctant to do that because it is explicitly made private.

Reply:
Take a look: https://github.com/test-kitchen/test-kitchen/pull/1679/commits/013b9d5075329140510cea72b1c086eb05b17d90